configure.ac

```diff
@@ -452,6 +452,7 @@ AC_SUBST([ODP_LIB_NAME])
 DX_HTML_FEATURE(ON)
 DX_PDF_FEATURE(OFF)
 DX_PS_FEATURE(OFF)
+DX_DOT_FEATURE(ON)
```

Comment:
Running `make doxygen-doc` on a system which doesn't have `dot` installed causes a lot of errors:
```
DXGEN  doc/application-api-guide/Doxyfile
sh: 1: dot: not found
error: Problems running dot: exit code=127, command='dot', arguments='"/home/nnn/dev/odp.git/doc/application-api-guide/output/html/api_2abi-default_2align_8h__incl.dot" -Tsvg -o "/home/nnn/dev/odp.git/doc/application-api-guide/output/html/api_2abi-default_2align_8h__incl.svg"'
sh: 1: dot: not found
error: Problems running dot: exit code=127, command='dot', arguments='"/home/nnn/dev/odp.git/doc/application-api-guide/output/latex/api_2abi-default_2align_8h__incl.dot" -Tpdf -o "/home/nnn/dev/odp.git/doc/application-api-guide/output/latex/api_2abi-default_2align_8h__incl.pdf"'
sh: 1: dot: not found
...
```

Adding `HAVE_DOT = $(HAVE_DOT)` to `doc/application-api-guide/Doxyfile` and `doc/platform-api-guide/Doxyfile` seems to fix this.

Reply:
But those include Doxyfile_common, where that assignment is added in this patch. When dot is not available, all I get is

```
$ ./bootstrap && ./configure && make doxygen-doc
...
checking for dot... no
configure: WARNING: dot not found - will not generate graphics for doxygen documentation
...
  DXGEN  doc/application-api-guide/Doxyfile
  DXGEN  doc/helper-guide/Doxyfile
  DXGEN  doc/platform-api-guide/Doxyfile

```
and

```
$ ./bootstrap && ./configure --enable-user-guides && make doxygen-doc
...
checking for dot... no
configure: WARNING: dot not found - continuing without dot graphics support
configure: error: cannot generate user guides without dot

```

Reply:
Different Doxygen version perhaps? I'm running Ubuntu 18.04 and Doxygen version is 1.8.13.